### PR TITLE
New README and other markdown

### PR DESCRIPTION
**Pull request by WilliamKelley**
This PR covers a bunch of markdown file changes, with wording and styling inspired and sometimes copied directly from repos like [Material-UI](https://github.com/mui-org/material-ui/).

Changed:
- Monorepo `README.md`
  - this also gets shown on NPM, so it has to be fairly public-oriented in terms of content presented.
  - uses a multicolor Spark monogram logo from the [experimental documentation website Figma page](https://www.figma.com/file/1DFL8ekapCuRe0cajNDMRI/PDS---Documentation-Website?node-id=66%3A442&viewport=603%2C475%2C0.17874965071678162) (see bottom of post for pic). I chose this instead of the more [official logos in Base](https://www.figma.com/file/w9lodWeivyS4GHdBNIYaJr/Spark-Design-System---Base?node-id=5%3A0&viewport=350%2C423%2C0.5379844903945923) because depending on the Light vs Dark theme of GitHub, one or the other will have barely any contrast, whereas the multicolor has more acceptable contrast all around (still not perfect though). And I don't know how (or if it's possible) to evaluate the page background to conditionally color an image/svg.
  - adds a bunch of cool badge (see bottom of post for pic) (see [shields.io](https://shields.io/), [simpleicons](https://simpleicons.org/), [github repo](https://github.com/badges/shields))

Added:
- `CONTRIBUTOR.md` -- linked to in the README. There's a ton of information that could fit, but hopefully it covers the broad strokes.
- ~~`CODE_OF_CONDUCT.md` -- this is an official adoption of the fairly standard [Contributor Covenant](https://www.contributor-covenant.org/)~~ _(Edit: Removed, see thread)_
- `LICENSE` -- just the MIT license, standard for our purpose

You can view the formatted markdown in Files Changed by clicking the 3 dots at the top right > View File.
Some asset links won't appear correctly because the asset isn't on the `main` branch yet; the data uri has to point to main because it has to be accessible from NPM's site, so it can't be an internal repo reference. Here's a preview where I temporarily changed the uri
image
image

alternatively there's a lighter multicolor that has more of unacceptable contrast on Light
image
image

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+### Open an Issue first
+
+Before working on any change besides chores, it is best to open an issue first and discuss it with the maintainers.
+
+## Get Up and Running
+
+There will be differences depending on if you are inside or outside of Prenda. See [Outside of Prenda?](##-outside-of-prenda) for our outlook on this.
+
+1. Inside of Prenda? Skip this step. Outside of Prenda? Fork the repository.
+2. Inside of Prenda? Clone the repository to your local machine. Outside of Prenda? Clone the fork to your local machine, add upstream remote, and synchronize your local with upstream. Else: goto 3. I
```
Comment (shadoath) [inline CONTRIBUTING.md:110]: This part is a bit confusing. I think we could simply go right to the headings `# Inside of Prenda` then list all steps in order followed by `# Outside of Prenda` with all steps. These could both have a `goto 3` at the end.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+2. The designer files a "New Component" Issue in this repository. The description should:
+  - Provide a link to the corresponding Figma page.
+  - Describe the motivation and use-case of component, if necessary.
+  - Enumerate descriptors like variants, sizes, etc.
+  - Describe component makeup, if the component has, or is, a composition pattern.
+3. An engineer assesses the description, communicates with the designer as necessary, and produces an initial written assessment:
+  - Formalize enumerated descriptors to props
+  - Suggest which components from [Material-UI](https://material-ui.com) can be custom-styled or modified to produce the new component.
+4. Label the issue with "ready" (to be worked on).
+5. A developer "claims" the issue (assigns themselves), create a new branch, begins work on implementation.
```
Comment (shadoath) [inline CONTRIBUTING.md:26]: **An engineer** and **A Developer** are both the same right? Would it be better to keep as one?

File: README.md [hunk 3]
```diff
@@ -1,51 +1,80 @@
-# PrendaSpark
+<p align="center">
+  
+<img width="128" src="https://github.com/prenda-school/prenda-spark/blob/main/public/img/spark-logo-multicolor-darker.svg" alt="Prenda Spark logo">
```
Comment (shadoath) [inline README.md:3]: I'm good with the multicolor dark Prenda Spark logo.

**Review by shadoath — COMMENTED** [review]
Nice work on this William. I was looking at the readme earlier feeling it needed some updates and here they are.

File: CONTRIBUTING.md (hunk 2 above)
Comment (WilliamKelley) [reply CONTRIBUTING.md:26]: Probably.

File: CONTRIBUTING.md (hunk 1 above)
Comment (WilliamKelley) [reply CONTRIBUTING.md:110]: Fair enough, I was trying to cut down on repetition, but i could tell it was getting confusing.

**Comment by WilliamKelley** [comment]
@shadoath Thanks for the feedback. I'll have limited time to work on this for a bit, so feel free to make changes as you see fit.

File: CONTRIBUTING.md (hunk 1 above)
Comment (WilliamKelley) [reply CONTRIBUTING.md]: Pushed a change for this

**Comment by seigler** [comment]
(my comment, copied from a Slack conversation about contributor covenant)
> IMHO we should not include a code of conduct in the repository. Prenda employees are already expected to behave professionally, and Prenda's existing HR processes apply when they fail to meet this standard. We don't have an outside-contributor-abuse problem now, but if we did we are already empowered to act unilaterally to resolve it since we maintain control of the repo ourselves.

If we do decide to make such a document, I recommend including Prenda leadership & legal in that process, and hosting it on the Prenda website, to be linked in our open-source projects where needed.

**Comment by WilliamKelley** [comment]
In regards to Seigler's concern and our followup discussion and Slack, the Contributor Covenant and official Code of Conduct has been removed from this PR as it's largely unnecessary for now. I revised the Code of Conduct section of CONTRIBUTOR.md to a small excerpt.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+    - Describe the before-and-after of the change.
+2. An engineer assesses the description, communicates with the designer as necessary, and produces a change plan.
+custom-styled or modified to produce the new component.
+4. Label the issue with "ready" (to be worked on).
+5. A engineer "claims" the issue (assigns themselves), create a new branch, begins work on implementation.
+
+### Implementation Work
+
+Requirements:
+- Spark component is a wrapper around, or is based on, a Material-UI component(s).
```
[the remark this answers is not in the document]
Comment (WilliamKelley) [reply CONTRIBUTING.md]: done

**Comment by WilliamKelley** [comment]
@awburgard @seigler Added a little intro for that section that goes into "Based on Material-UI components". And I condensed the example codes and co-located most of the bullets for hopefully increased readability.

**Pull request merged**